cvxopt: 1.1.7 -> 1.1.9, add optional extensions

timokau opened the pull request:
###### Motivation for this change

- update cvxopt to the latest version
- add support for optional extensions: glpk, gsl, fftw
- move out of `python-packages`
- run tests

Maintainer: @edwtjo 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

timokau committed:
pythonPackages.cvxopt: 1.1.7 -> 1.1.9

Also adds support for the optional extensions glpk, gsl, fftw and
enables tests.

dotlambda commented on pkgs/development/python-modules/cvxopt/default.nix:
```diff
@@ -0,0 +1,71 @@
+    export CVXOPT_FFTW_INC_DIR=${fftw.dev}/include
+  '';
+  checkPhase = ''
+    python -m unittest discover -s tests
```
Replace `python` by `${python.interpreter}`

dotlambda commented on pkgs/development/python-modules/cvxopt/default.nix:
```diff
@@ -0,0 +1,71 @@
+  buildInputs = [
+    openblasCompat
+    suitesparse
+  ] ++ stdenv.lib.optionals withGlpk [
```
`stdenv.lib` can be replaced by `lib` everywhere

dotlambda commented on pkgs/development/python-modules/cvxopt/default.nix:
```diff
@@ -0,0 +1,71 @@
+{ stdenv
```
Use `lib` here

dotlambda reviewed (changes requested):
Please add a little bit of white space, i.e. blank lines.

dotlambda commented:
I don't see nothing moved out of pythonPackages.

timokau commented:
I addressed your comments. I don't know what you mean with the last one: `cvxopt` was previously defined in `python-packages.nix`, now is in its own file in `python-modules`.

dotlambda commented:
The commit message needs to be amended here as well.

@GrahamcOfBorg build python2.pkgs.cvxopt python3.pkgs.cvxopt

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38780&attempt_id=3a47a286-5208-4c07-9bfe-e6318af11992)

Attempted: python2.pkgs.cvxopt, python3.pkgs.cvxopt

<details><summary>Partial log (click to expand)</summary><p>

```
 3: -2.4676e+02 -2.4620e+02  4e+00  3e-01  5e-15
 4: -2.4543e+02 -2.4547e+02  1e+00  8e-02  6e-15
 5: -2.4515e+02 -2.4527e+02  5e-01  2e-02  9e-15
 6: -2.4510e+02 -2.4517e+02  1e-01  3e-03  1e-14
 7: -2.4512e+02 -2.4515e+02  3e-02  3e-04  3e-14
 8: -2.4514e+02 -2.4514e+02  6e-03  6e-05  4e-14
 9: -2.4514e+02 -2.4514e+02  1e-03  7e-06  8e-14
10: -2.4514e+02 -2.4514e+02  2e-04  1e-06  1e-13
11: -2.4514e+02 -2.4514e+02  1e-05  4e-16  3e-13
Optimal solution found.
```
</p></details>

timokau commented:
Done.

dotlambda commented on pkgs/development/python-modules/cvxopt/default.nix:
```diff
@@ -0,0 +1,78 @@
+    glpk
+  ] ++ lib.optionals withGsl [
+    gsl
+  ];
```
You should be able to leave `buildInputs` out completely because they are all mentioned in `preConfigure`.

dotlambda commented:
@GrahamcOfBorg build python2.pkgs.cvxopt python3.pkgs.cvxopt

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38780&attempt_id=aa96d5e7-0374-4ea0-82b5-c606c36caec2)

Attempted: python2.pkgs.cvxopt, python3.pkgs.cvxopt

<details><summary>Partial log (click to expand)</summary><p>

```
 3: -2.4676e+02 -2.4620e+02  4e+00  3e-01  5e-15
 4: -2.4543e+02 -2.4547e+02  1e+00  8e-02  8e-15
 5: -2.4515e+02 -2.4527e+02  5e-01  2e-02  1e-14
 6: -2.4510e+02 -2.4517e+02  1e-01  3e-03  2e-14
 7: -2.4512e+02 -2.4515e+02  3e-02  3e-04  4e-14
 8: -2.4514e+02 -2.4514e+02  6e-03  6e-05  5e-14
 9: -2.4514e+02 -2.4514e+02  1e-03  7e-06  1e-13
10: -2.4514e+02 -2.4514e+02  2e-04  1e-06  2e-13
11: -2.4514e+02 -2.4514e+02  1e-05  4e-16  4e-13
Optimal solution found.
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38780&attempt_id=f4de487c-0d0d-4a8e-865e-fea22e5863c9)

Attempted: python2.pkgs.cvxopt, python3.pkgs.cvxopt

<details><summary>Partial log (click to expand)</summary><p>

```
 3: -2.4676e+02 -2.4620e+02  4e+00  3e-01  5e-15
 4: -2.4543e+02 -2.4547e+02  1e+00  8e-02  6e-15
 5: -2.4515e+02 -2.4527e+02  5e-01  2e-02  9e-15
 6: -2.4510e+02 -2.4517e+02  1e-01  3e-03  1e-14
 7: -2.4512e+02 -2.4515e+02  3e-02  3e-04  3e-14
 8: -2.4514e+02 -2.4514e+02  6e-03  6e-05  4e-14
 9: -2.4514e+02 -2.4514e+02  1e-03  7e-06  8e-14
10: -2.4514e+02 -2.4514e+02  2e-04  1e-06  1e-13
11: -2.4514e+02 -2.4514e+02  1e-05  4e-16  3e-13
Optimal solution found.
```
</p></details>
